### PR TITLE
Move raise AssertionError to its own function

### PR DESCRIPTION
**Pull request by cdonovick**

### CODE REVIEW

File: fault/subprocess_run.py [hunk 1]
```diff
@@ -75,6 +75,11 @@ def error_detected(text, err_str):
         raise Exception(f'Invalid err_str: {err_str}.')
 
 
+def _throw(args, cwd, env, disp_type, err_str,
+           chk_ret_code, shell, use_fault_cfg):
+    raise AssertionError()
```
Comment (leonardt) [inline fault/subprocess_run.py:80]: Can we add a short comment noting why we do this?

**Review by leonardt — APPROVED** [review]
LGTM

File: fault/subprocess_run.py (hunk 1 above)
Comment (cdonovick) [reply fault/subprocess_run.py:80]: Sure, will do

Comment (cdonovick) [reply fault/subprocess_run.py:80]: I added the comment at the call site as opposed to here so we don't need to read the comment on every failed test.

**Pull request merged**